src/AgentImplementation.sol

```diff
@@ -266,14 +266,9 @@ contract AgentImplementation is IAgent, ERC721Holder, ERC1155Holder {
             }
 
             // Set callback who should enter one-time `executeByCallback`
-            if (logics[i].callback != address(0)) {
-                bytes32 callback = bytes32(bytes20(logics[i].callback));
-                if (chargeFeeOnCallback) {
-                    // Set the least significant bit
-                    _callbackWithCharge = callback | _CHARGE_MASK;
-                } else {
-                    _callbackWithCharge = callback;
-                }
+            address callback = logics[i].callback;
```

Comment:
Do we need an extra memory copy here? Seems using `logics[i].callback` is enough.

Reply:
It used twice here so I still assign it separately. But considering applying callback should be a minor case, maybe using it directly is fine.